agent/hcp/deps.go

```diff
@@ -25,18 +24,28 @@ type Deps struct {
 	Sink     metrics.MetricSink
 }
 
-func NewDeps(cfg config.CloudConfig, logger hclog.Logger, nodeID types.NodeID) (Deps, error) {
+func NewDeps(cfg config.CloudConfig, logger hclog.Logger) (Deps, error) {
+	logger = logger.Named("sink")
+	ctx := context.Background()
+	ctx = hclog.WithContext(ctx, logger)
+
 	client, err := hcpclient.NewClient(cfg)
 	if err != nil {
-		return Deps{}, fmt.Errorf("failed to init client: %w:", err)
+		return Deps{}, fmt.Errorf("failed to init client: %w", err)
 	}
 
 	provider, err := scada.New(cfg, logger.Named("scada"))
 	if err != nil {
 		return Deps{}, fmt.Errorf("failed to init scada: %w", err)
 	}
 
-	sink := sink(client, &cfg, logger.Named("sink"), nodeID)
+	metricsClient, err := hcpclient.NewMetricsClient(ctx, &cfg)
+	if err != nil {
+		logger.Error("failed to init metrics client", "error", err)
+		return Deps{}, fmt.Errorf("failed to init metrics client: %w", err)
+	}
+
+	sink := sink(ctx, client, metricsClient, cfg)
```

Comment:
Pulling out the metrics client instantion allows us to pass the configuration in as a CloudConfig instead of an interface as is done for the NewMetricsClient. This allows us to actually access configuration values stored in the config struct.

Reply:
We could honestly even remove the interface for the MetricsClient. In my initial implementation, it felt like the easiest way to test, but I have some alternate ideas. We could do that in a follow up PR maybe! Ideally we also cleanup / refactor and test the config object (a rabbit hole :p)!